Tests/JetStreamTests/Integration/JetStreamTests.swift

```diff
@@ -310,4 +312,107 @@ class JetStreamTests: XCTestCase {
+        XCTAssertEqual(msg.payload, "50".data(using: .utf8)!)
+
+        // get by sequence and subject
+        msg = try await stream.getMsg(sequence: 50, subject: "foo.B")
+        // msg with sequence 50 is on subject foo.A, so we expect the next message which should be on foo.B
+        XCTAssertEqual(msg.payload, "51".data(using: .utf8)!)
+        XCTAssertEqual(msg.headers, hm)
+
+        // get first message from a subject
+        msg = try await stream.getMsg(firstForSubject: "foo.A")
+        XCTAssertEqual(msg.payload, "2".data(using: .utf8)!)
+        XCTAssertEqual(msg.headers, hm)
+
+        // get last message from subject
+        msg = try await stream.getMsg(lastForSubject: "foo.B")
+        XCTAssertEqual(msg.payload, "99".data(using: .utf8)!)
+        XCTAssertEqual(msg.headers, hm)
+
+        // message not found
+        do {
+            _ = try await stream.getMsg(sequence: 200)
+        } catch let err as JetStreamError {
+            XCTAssertEqual(err.errorCode, .noMessageFound)
+        }
+    }
+
+    func testGetMsgDirect() async throws {
+        let bundle = Bundle.module
+        natsServer.start(
+            cfg: bundle.url(forResource: "jetstream", withExtension: "conf")!.relativePath)
+        logger.logLevel = .debug
+
+        let client = NatsClientOptions().url(URL(string: natsServer.clientURL)!).build()
+        try await client.connect()
+
+        let ctx = JetStreamContext(client: client)
+
+        let cfg = StreamConfig(name: "STREAM", subjects: ["foo.*"], allowDirect: true)
+        let stream = try await ctx.createStream(cfg: cfg)
+
+        var hm = NatsHeaderMap()
+        hm[try! NatsHeaderName("foo")] = NatsHeaderValue("bar")
+        hm.append(try! NatsHeaderName("foo"), NatsHeaderValue("baz"))
+        hm[try! NatsHeaderName("key")] = NatsHeaderValue("val")
+        for i in 1...100 {
+            let msg = "\(i)".data(using: .utf8)!
+            let subj = i % 2 == 0 ? "foo.A" : "foo.B"
+            let ack = try await ctx.publish(subj, message: msg, headers: hm)
+            _ = try await ack.wait()
+        }
+
+        // get by sequence
+        var msg = try await stream.getMsgDirect(sequence: 50)
+        XCTAssertEqual(msg.payload, "50".data(using: .utf8)!)
+
+        // get by sequence and subject
+        msg = try await stream.getMsgDirect(sequence: 50, subject: "foo.B")
+        // msg with sequence 50 is on subject foo.A, so we expect the next message which should be on foo.B
+        XCTAssertEqual(msg.payload, "51".data(using: .utf8)!)
+
+        // get first message from a subject
+        msg = try await stream.getMsgDirect(firstForSubject: "foo.A")
+        XCTAssertEqual(msg.payload, "2".data(using: .utf8)!)
+
+        // get last message from subject
+        msg = try await stream.getMsgDirect(lastForSubject: "foo.B")
+        XCTAssertEqual(msg.payload, "99".data(using: .utf8)!)
+
+        // message not found
+        do {
```

Comment:
this is an example of the argument I make above. if we think 404 is going to be rare this would be fine. otherwise probably an optional would be better.